invenio_rdm_records/oaiserver/services/services.py

```diff
@@ -130,7 +131,7 @@ def search(self, identity, params):
         """Perform search over OAI sets."""
         self.require_permission(identity, "read")
 
-        search_params = self._get_search_params(params)
+        search_params = get_search_params(params)
```

Comment:
This has not been tested: you are missing to pass the `service search config` as first param.